File: tests/fixtures/vulnerable/ssrf.php
<?php
function pull() {
  // EXPECT-FLOW ssrf
  return file_get_contents($_GET['url']);
}
function pullCurl() {
  $ch = curl_init();
  // EXPECT-FLOW ssrf
  curl_setopt($ch, CURLOPT_URL, $_POST['endpoint']);
  return curl_exec($ch);
}
function safe() {
  return file_get_contents('https://api.example.com/x');  // SAFE
}

File: tests/fixtures/vulnerable/ssrf.py
import requests, urllib.request
from flask import request

def fetch_profile():
    # EXPECT-FLOW ssrf
    return requests.get(request.args["url"]).text

def fetch_raw():
    # EXPECT-FLOW ssrf
    return urllib.request.urlopen(request.form["endpoint"]).read()

def fetch_fixed():
    return requests.get("https://api.example.com/status").text       # SAFE
